src/libraries/System.Security.Cryptography.Encoding/src/System/Security/Cryptography/Oid.cs

```diff
@@ -86,17 +96,40 @@ public static Oid FromOidValue(string oidValue, OidGroup group)
             }
             set
             {
-                _friendlyName = value;
+                // If _friendlyName has not been set, permit it to be set once, or to
+                // the same value for "initialize once" behavior.
+                if (_friendlyName != null && !_friendlyName.Equals(value, StringComparison.Ordinal))
+                {
+                    throw new PlatformNotSupportedException(SR.Cryptography_Oid_SetOnceFriendlyName);
+                }
+
                 // If we can find the matching OID value, then update it as well
-                if (_friendlyName != null)
+                if (value != null)
                 {
                     // If FindOidInfo fails, we return a null String
-                    string? oidValue = OidLookup.ToOid(_friendlyName, _group, fallBackToAllGroups: true);
+                    string? oidValue = OidLookup.ToOid(value, _group, fallBackToAllGroups: true);
                     if (oidValue != null)
                     {
-                        _value = oidValue;
+                        // If the OID value has not been initialized, set it
+                        // to the lookup value.
+                        if (_value == null)
+                        {
+                            _value = oidValue;
+                        }
+
+                        // The friendly name resolves to an OID value other than the
+                        // current one, which is not permitted under "initialize once"
+                        // behavior.
+                        else if (!_value.Equals(oidValue, StringComparison.Ordinal))
+                        {
+                            throw new PlatformNotSupportedException(SR.Cryptography_Oid_SetOnceValue);
+                        }
                     }
                 }
+
+                // Ensure we don't mutate _friendlyName until we are sure we can
+                // set _value if we are going to.
```

Comment:
While I understand what the comment means, I'm not sure what "set _value" means specifically.

```suggestion
                // accept the value.
```

?

Reply:
Oh, I get it now.  `_value`.  Duh.

Reply:
It means the `_value` field. I`ll fix the wording but what it was trying to avoid was 1. Set _friendlyName, looking new OID value, oh wait, the OID value changed, and now we've permitted a change in the friendly name and already written it to the field.